presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateDoublePercentileArrayAggregations.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.DigestAndPercentileArrayState;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.SqlType;
+import com.google.common.collect.ImmutableList;
+import io.airlift.stats.QuantileDigest;
+
+import java.util.List;
+
+import static com.facebook.presto.operator.aggregation.ApproximateLongPercentileArrayAggregations.addRawValueToDigest;
+import static com.facebook.presto.operator.aggregation.ApproximateLongPercentileArrayAggregations.addWeightedRawValueToDigest;
+import static com.facebook.presto.operator.aggregation.ApproximateLongPercentileArrayAggregations.initializeDigest;
+import static com.facebook.presto.operator.aggregation.ApproximateLongPercentileArrayAggregations.initializePercentilesArray;
+import static com.facebook.presto.operator.aggregation.LongDoubleConverterUtil.doubleToSortableLong;
+import static com.facebook.presto.operator.aggregation.LongDoubleConverterUtil.sortableLongToDouble;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+
+@AggregationFunction("approx_percentile")
+public final class ApproximateDoublePercentileArrayAggregations
+{
+    public static final InternalAggregationFunction DOUBLE_APPROXIMATE_PERCENTILE_ARRAY_AGGREGATION = new AggregationCompiler().generateAggregationFunction(ApproximateDoublePercentileArrayAggregations.class, new ArrayType(DOUBLE), ImmutableList.<Type>of(DOUBLE, new ArrayType(DOUBLE)));
+    public static final InternalAggregationFunction DOUBLE_APPROXIMATE_PERCENTILE_ARRAY_WEIGHTED_AGGREGATION = new AggregationCompiler().generateAggregationFunction(ApproximateDoublePercentileArrayAggregations.class, new ArrayType(DOUBLE), ImmutableList.<Type>of(DOUBLE, BIGINT, new ArrayType(DOUBLE)));
+
+    private ApproximateDoublePercentileArrayAggregations() {}
+
+    @InputFunction
+    public static void input(DigestAndPercentileArrayState state, @BlockPosition @SqlType(StandardTypes.DOUBLE) Block block, @BlockIndex int index, @SqlType("array<double>") Block percentilesArrayBlock)
```

Comment:
Now that the nulls are gone, we can just have the annotated functions take the raw types directly (e.g. double instead of block and index). That way we can write most if not all of the double percentile functions directly in terms of the long percentile functions.